python/test/unit/conftest.py

```diff
@@ -26,20 +28,50 @@ def fresh_triton_cache():
             os.environ.pop("TRITON_CACHE_DIR", None)
 
 
-@pytest.fixture
-def fresh_knobs(request, monkeypatch):
+def _fresh_knobs_impl(monkeypatch, skipped_attr: Optional[Set[str]] = None):
     from triton import knobs
+
+    if skipped_attr is None:
+        skipped_attr = set()
+
     knobs_map = {
         name: knobset
         for name, knobset in knobs.__dict__.items()
-        if isinstance(knobset, knobs.base_knobs) and knobset != knobs.base_knobs
+        if isinstance(knobset, knobs.base_knobs) and knobset != knobs.base_knobs and name not in skipped_attr
     }
-    try:
+
+    def fresh_function():
         for name, knobset in knobs_map.items():
             setattr(knobs, name, knobset.copy().reset())
             for knob in knobset.knob_descriptors.values():
                 monkeypatch.delenv(knob.key, raising=False)
-        yield knobs
-    finally:
+        return knobs
+
+    def reset_function():
         for name, knobset in knobs_map.items():
             setattr(knobs, name, knobset)
+
+    return fresh_function, reset_function
+
+
+@pytest.fixture
+def fresh_knobs(monkeypatch):
+    fresh_function, reset_function = _fresh_knobs_impl(monkeypatch)
+    try:
+        yield fresh_function()
+    finally:
+        reset_function()
+
+
+@pytest.fixture
+def fresh_knobs_except_libraries(monkeypatch):
+    """
+    A variant of `fresh_knobs` that keeps library path
+    information from the environment as these may be
+    needed to successfully compile kernels.
+    """
+    fresh_function, reset_function = _fresh_knobs_impl({"build", "nvidia", "amd"})
```

Comment:
Looks like you forgot the `monkeypatch` arg here

Reply:
Forgot to copy it over from the internal copy I was testing. Thanks! Will update.